### DISCUSSION TITLE
Add functionality in Windows to specify Monitor ID, Device ID, Short Monitor Name as display output

### DISCUSSION THREAD

**Discussion by HisshouBuraiKen** [LizardByte/community#287] [Sunshine]
The \\.\DISPLAY# id will sometimes change numbers when adding a new monitor, tweaking settings, etc. In Windows.  However, Windows exposes other immutable display properties like the Monitor ID (And Short Monitor ID) as well as a device ID that can easily be found from freeware like MultiMonitorTool.

It seems like it would be a simple, helpful addition to be able to reference these other unique display IDs to make using external monitors and dummy plugs work more consistently without additional software or unnecessary scripting.

**Comment by ReenigneArcher** [comment]
We are currently encouraging the use of device names since they are less likely to change between sessions. While specific to audio devices, see here for some context. https://github.com/LizardByte/Sunshine/pull/1246

Someone can correct me if I'm wrong, but I believe the same would be true for displays. Essentially everyone had to reconfigure their audio devices every time they did a Windows update.

**Comment by cgutman** [comment]
Yeah, we should definitely implement this for display devices like we did for audio devices.

In addition to the "adding a new monitor" case, the driver upgrade case causes _all_ displays on the GPU to be removed and created again with new names. If you do this while streaming, Sunshine will end up in an infinite loop trying to recreate the display because it's looking for the old display names and keeps cycling through all known displays trying to find them again.

**Reply by OGTK423** [reply]
Is this still a planned feature to be implemented?

**Comment by ld892012** [comment]
> Is this still a planned feature to be implemented?

It seems to be in active development : 
[Pull#2032](https://github.com/LizardByte/Sunshine/pull/2032)

**Reply by ReenigneArcher** [reply]
Correct, and a good portion of that is getting turned into a separate library: https://github.com/LizardByte/libdisplaydevice
